third-party/linshare/src/main/java/org/apache/james/linshare/LinshareConfiguration.java

```diff
@@ -30,67 +30,82 @@
 
     public static class Builder {
         @FunctionalInterface
         public interface RequireUrl {
-            RequireAuthorizationToken url(URL url);
+            RequireBasicAuthorization url(URL url);
 
-            default RequireAuthorizationToken urlAsString(String url) throws MalformedURLException {
+            default RequireBasicAuthorization urlAsString(String url) throws MalformedURLException {
                 return url(new URL(url));
             }
         }
 
-        public interface RequireAuthorizationToken {
-            ReadyToBuild authorizationToken(AuthorizationToken token);
+        public interface RequireBasicAuthorization {
+            ReadyToBuild basicAuthorization(String username, String password);
         }
 
         public static class ReadyToBuild {
             private final URL url;
-            private final AuthorizationToken token;
+            private final String username;
+            private final String password;
 
-            ReadyToBuild(URL url, AuthorizationToken token) {
+            ReadyToBuild(URL url, String username, String password) {
                 this.url = url;
-                this.token = token;
+                this.username = username;
+                this.password = password;
             }
 
             public LinshareConfiguration build() {
-                return new LinshareConfiguration(url, token);
+                return new LinshareConfiguration(url, username, password);
             }
         }
     }
 
     public static Builder.RequireUrl builder() {
-        return url -> credential -> new Builder.ReadyToBuild(url, credential);
+        return url -> (username, password) -> new Builder.ReadyToBuild(url, username, password);
     }
 
     public static LinshareConfiguration from(Configuration configuration) throws MalformedURLException {
         return builder()
-            .urlAsString(configuration.getString(URL, null))
-            .authorizationToken(new AuthorizationToken(configuration.getString(TOKEN, null)))
+            .urlAsString(configuration.getString(URL_PROPERTY, null))
+            .basicAuthorization(
+                configuration.getString(USER_NAME_PROPERTY, null),
+                configuration.getString(PASSWORD_PROPERTY, null))
             .build();
     }
 
     private final URL url;
-    private final AuthorizationToken token;
+    private final String username;
+    private final String password;
 
     @VisibleForTesting
-    LinshareConfiguration(URL url, AuthorizationToken token) {
-        Preconditions.checkNotNull(url, "'" + URL + "' can not be null");
-        Preconditions.checkNotNull(token, "'" + TOKEN + "' can not be null");
+    LinshareConfiguration(URL url, String username, String password) {
+        Preconditions.checkNotNull(url, "'%s' can not be null", URL_PROPERTY);
+        Preconditions.checkNotNull(username, "'%s' can not be null", USER_NAME_PROPERTY);
```

Comment:
it's `USERNAME` or you need to change `username` to `userName`

Reply:
You mean `USER_NAME_PROPERTY` should be `USERNAME_PROPERTY` right?